vaccine/vaccine_reg_ussd.py

```diff
@@ -0,0 +1,363 @@
+from datetime import date
+from enum import Enum
+
+from vaccine.base_application import BaseApplication
+from vaccine.states import (
+    Choice,
+    ChoiceState,
+    EndState,
+    ErrorMessage,
+    FreeText,
+    MenuState,
+)
+from vaccine.utils import luhn_checksum
+
+MAX_AGE = 122
+
+
+class Application(BaseApplication):
+    START_STATE = "state_age_gate"
+
+    class ID_TYPES(Enum):
+        rsa_id = "RSA ID Number"
+        passport = "Passport Number"
+        asylum_seeker = "Asylum Seeker Permit number"
+        refugee = "Refugee Number Permit number"
+
+    async def state_age_gate(self):
+        return MenuState(
+            self,
+            question="\n".join(
+                [
+                    "VACCINE REGISTRATION",
+                    "The SA Department of Health thanks you for helping to defeat the "
+                    "coronavirus!",
+                    "",
+                    "Are you 40 years or older?",
+                ]
+            ),
+            choices=[
+                Choice("state_identification_type", "Yes"),
+                Choice("state_under40_notification", "No"),
+            ],
+            error="Self registration is currently only available to those 40 years or "
+            "older. Please tell us if you are 40 years of age or older?",
+        )
+
+    async def state_under40_notification(self):
+        return ChoiceState(
+            self,
+            question="Self registration is only available to those 40 years or older. "
+            "Can we notifify you by SMS on this number when this changes?",
```

Comment:
typo on `notifify`, but not an issue since we need to update this with the official copy when it's ready